app/ts/components/simulationExplaining/customExplainers/CatchAllVisualizer.tsx

```diff
@@ -119,7 +121,7 @@ export function CatchAllVisualizer(param: TransactionImportanceBlockParams) {
 			{ /* contract creation */}
 			{ param.simTx.transaction.to !== undefined ? <></> : <>
 				<div class = 'log-cell' style = 'justify-content: left; display: grid;'>
-					<p class = 'paragraph'> The transaction deploys a contract </p>
+					<p class = 'paragraph'> { `A contract is deployed to address ${ addressString(getDeployedContractAddress(param.simTx.transaction.from.address, param.simTx.transaction.nonce)) }` }</p>
```

Comment:
instead of printing address directly, make it a SmallAddress component and get the addresses addressbook entry from local storage